File: 1544-count-good-nodes-in-binary-tree/1544-count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    void helper(TreeNode *root, int maxi, int &ans)
    {
        if(!root)
            return;

        if(root->val>=maxi)
        {    
            ans++;
            maxi = root->val;
        }

        helper(root->right, maxi, ans);
        helper(root->left, maxi, ans);
    }
public:
    int goodNodes(TreeNode* root) {
        int ans = 0;
        helper(root, INT_MIN, ans);
        return ans;
    }
};

File: 1544-count-good-nodes-in-binary-tree/1544-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    


    def goodNodes(self, root: TreeNode) -> int:
        def helper(root, count, maxval):
            if root.val >= maxval:
                count += 1

            if root.left:
                count = helper(root.left, count, max(root.val, maxval))

            if root.right:
                count = helper(root.right, count, max(root.val, maxval))

            return count

        return helper(root, 0, root.val)
            